mathesar/views/api.py

```diff
@@ -119,6 +119,32 @@ def type_suggestions(self, request, pk=None):
         col_types = get_table_column_types(table)
         return Response(col_types)
 
+    @action(methods=['post'], detail=True)
+    def preview(self, request, pk=None):
+        table = self.get_object()
+        serializer = TablePreviewSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors)
+        else:
+            columns = serializer.validated_data["columns"]
+
+        if not len(columns) == len(set([col["name"] for col in columns])):
+            raise ValidationError("Column names must be distinct")
```

Comment:
This error might not always mean that column names are distinct, it could be that there are fewer columns in the list. Right?

Reply:
I think it should work, since very column in `columns` is required to include a `"name"` key.  So, this should just check that we don't lose any `columns` when taking the set of the list of names of each column in `columns`.  I'll rewrite it to be clearer.

Reply:
You might also just move lines 141-142 above this `if not` statement. That would make sure the number of fields is checked before checking against the cast set of names.